tests/tracing/test_misc.py

```diff
@@ -509,3 +509,34 @@ def test_transaction_not_started_warning(sentry_init):
         "The transaction will not be sent to Sentry. To fix, start the transaction by"
         "passing it to sentry_sdk.start_transaction."
     )
+
+
+def test_span_set_data(sentry_init, capture_events):
```

Comment:
I would also add `update_data` to the name of the test. And, perhaps add a test which only checks `update_data` in isolation

Reply:
done